app/src/main/java/com/jerboa/ui/components/comment/edit/CommentEditViewModel.kt

```diff
@@ -27,24 +33,33 @@ class CommentEditViewModel : ViewModel() {
 
     fun editComment(
         content: String,
-        ctx: Context,
         navController: NavController,
         focusManager: FocusManager,
         account: Account,
         personProfileViewModel: PersonProfileViewModel,
         postViewModel: PostViewModel,
     ) {
-        editCommentRoutine(
-            commentView = commentView,
-            loading = loading,
-            content = content,
-            ctx = ctx,
-            scope = viewModelScope,
-            navController = navController,
-            focusManager = focusManager,
-            account = account,
-            personProfileViewModel = personProfileViewModel,
-            postViewModel = postViewModel,
-        )
+        viewModelScope.launch {
+            commentView.value?.also { cv ->
+                val form = EditComment(
+                    content = content,
+                    comment_id = cv.comment.id,
+                    auth = account.jwt,
+                )
+
+                editCommentRes = ApiState.Loading
+                editCommentRes =
+                    apiWrapper(
+                        API.getInstance().editComment(form),
+                    )
+                focusManager.clearFocus()
+
+                // Update all the views which might have your comment
+                personProfileViewModel.updateComment(cv)
+                postViewModel.updateComment(cv)
```

Comment:
Its necessary because the initialize sets the comment_id. You'll notice the CommentView is optional, and the edit screen always needs to be initialized first.

Reply:
what i meant was

```
when (val res = editCommentRes) {
    is ApiState.Success -> {
        personProfileViewModel.updateComment(res.data.comment_view) // <- instead of cv
        postViewModel.updateComment(res.data.comment_view) // <- instead of cv
    }
    else -> {}
}
```

The other view models do this, it was missed in this ig.

Reply:
Ooooooh I see. Sry about that. I'll get it now.